packages/vega-parser/src/parsers/guides/axis-labels.js

```diff
@@ -40,30 +41,65 @@ export default function(spec, config, userEncode, dataRef, size, band) {
     offset: extendOffset(band.offset, _('labelOffset'))
   };
 
-  if (isXAxis) {
-    align = labelAlign || (flushOn
-      ? flushExpr(scale, flush, '"left"', '"right"', '"center"')
-      : 'center');
-    baseline = labelBaseline || (orient === Top ? 'bottom' : 'top');
-    offset = !labelAlign;
-  } else {
-    align = labelAlign || (orient === Right ? 'left' : 'right');
-    baseline = labelBaseline || (flushOn
-      ? flushExpr(scale, flush, '"top"', '"bottom"', '"middle"')
-      : 'middle');
-    offset = !labelBaseline;
-  }
+  if (isSignal(orient)) {
+    align = labelAlign ||
+      {
+        signal: `(${orient.signal}) === "${Top}" || (${orient.signal}) === "${Bottom}"`
+          + ' ? '
+          + `(${flushOn ? flushExpr(scale, flush, '"left"', '"right"', '"center"').signal : "'center'"})`
+          + ' : '
+          + `(${orient.signal}) === "${Right}" ? "left" : "right"`
+      };
 
-  offset = offset && flushOn && flushOffset
-    ? flushExpr(scale, flush, '-(' + flushOffset + ')', flushOffset, 0)
-    : null;
+    baseline = labelBaseline ||
+      {
+        signal: `(${orient.signal}) === "${Top}" ? "bottom"`
+          + ' : '
```

Comment:
Not sure why we don't append `:` in the previous line?

Reply:
fixed